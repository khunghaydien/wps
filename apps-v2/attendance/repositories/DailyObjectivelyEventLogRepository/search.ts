import Api from '@apps/commons/api';

import {
  convert,
  DailyObjectivelyEventLogRecord,
} from '../models/DailyObjectivelyEventLog';
import { IDailyObjectivelyEventLogRepository } from '@attendance/domain/models/DailyObjectivelyEventLog';

export type Response = {
  dailyRecordList: DailyObjectivelyEventLogRecord[];
};

const search: IDailyObjectivelyEventLogRepository['search'] = async ({
  employeeId,
  startDate,
  endDate,
}) => {
  const response: Response = await Api.invoke({
    path: '/att/daily-objectively-event-log/get',
    param: {
      empId: employeeId,
      startDate,
      endDate,
    },
  });
  return convert(response.dailyRecordList);
};

export default search;
