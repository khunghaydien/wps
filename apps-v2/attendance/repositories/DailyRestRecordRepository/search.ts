import Api from '@apps/commons/api';

import {
  convert,
  DailyRestRecord,
} from '@attendance/repositories/models/DailyRestRecord';

import { IDailyRestRecordRepository } from '@attendance/domain/models/DailyRestRecord';

export type Response = {
  dailyRestList: DailyRestRecord[];
};

const search: IDailyRestRecordRepository['search'] = async ({
  startDate,
  endDate,
  employeeId,
}) => {
  const result: Response = await Api.invoke({
    path: '/att/daily-rest/get',
    param: {
      startDate,
      endDate,
      empId: employeeId,
    },
  });
  return convert(result.dailyRestList);
};

export default search;
