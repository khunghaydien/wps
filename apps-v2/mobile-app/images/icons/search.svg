<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14.758 14.784">
  <defs>
    <style>
      .cls-1 {
        fill: #074b69;
      }
    </style>
  </defs>
  <g id="icons_utility_search" data-name="icons/utility/search" transform="translate(-0.615 -0.615)">
    <path id="Mask" class="cls-1" d="M15.261,13.938,11.138,9.846A5.814,5.814,0,1,0,.646,7.015a5.819,5.819,0,0,0,9.2,4.154l4.092,4.092a.446.446,0,0,0,.646,0l.646-.646a.516.516,0,0,0,.031-.677ZM2.461,6.43A3.969,3.969,0,1,1,6.43,10.4,3.979,3.979,0,0,1,2.461,6.43Z"/>
  </g>
</svg>
