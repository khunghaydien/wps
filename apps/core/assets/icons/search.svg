<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.781 10.7364L8.934 7.92177C9.61019 7.06985 9.98038 6.015 9.985 4.92696C9.96643 3.61388 9.43227 2.36086 8.49801 1.43887C7.56376 0.516879 6.30449 0 4.9925 0C3.68051 0 2.42124 0.516879 1.48699 1.43887C0.55273 2.36086 0.0185668 3.61388 0 4.92696C0.0109142 5.83653 0.26968 6.72586 0.748395 7.49905C1.22711 8.27224 1.90761 8.89997 2.71652 9.31454C3.52543 9.7291 4.43205 9.91478 5.33863 9.85156C6.24521 9.78833 7.11735 9.47859 7.861 8.95574L10.721 11.7834C10.8623 11.9222 11.0525 12 11.2505 12C11.4485 12 11.6387 11.9222 11.78 11.7834C11.8495 11.7152 11.9048 11.6338 11.9426 11.5439C11.9803 11.4541 11.9998 11.3576 11.9999 11.2601C12 11.1626 11.9807 11.0661 11.9431 10.9762C11.9055 10.8863 11.8504 10.8048 11.781 10.7364ZM4.993 8.38321C4.0723 8.3872 3.18758 8.02562 2.5328 7.37773C1.87802 6.72984 1.50661 5.84852 1.5 4.92696C1.54787 4.03152 1.93697 3.18854 2.58715 2.57164C3.23734 1.95473 4.09912 1.61087 4.995 1.61087C5.89088 1.61087 6.75266 1.95473 7.40285 2.57164C8.05303 3.18854 8.44213 4.03152 8.49 4.92696C8.48339 5.84921 8.11143 6.73115 7.45579 7.37915C6.80014 8.02716 5.91439 8.38827 4.993 8.38321Z" fill="#AFADAB"/>
</svg>
