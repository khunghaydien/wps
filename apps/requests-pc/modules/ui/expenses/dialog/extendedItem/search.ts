import { Reducer } from 'redux';

import { EISearchObj } from '../../../../../../domain/models/exp/ExtendedItem';

export const ACTIONS = {
  SET: 'MODULES/EXPENSES/DIALOG/EI_LOOKUP/LIST/SET',
  CLEAR: 'MODULES/EXPENSES/DIALOG/EI_LOOKUP/LIST/CLEAR',
};

export const actions = {
  set: (item: EISearchObj) => ({
    type: ACTIONS.SET,
    payload: item,
  }),
  clear: () => ({
    type: ACTIONS.CLEAR,
  }),
};

//
// Reducer
//
const initialState = {
  extendedItemLookupId: null,
  extendedItemCustomId: null,
  name: null,
  idx: null,
  target: null,
  hintMsg: null,
};

export default ((state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.SET:
      return action.payload;
    case ACTIONS.CLEAR:
      return initialState;
    default:
      return state;
  }
}) as Reducer<EISearchObj, any>;
