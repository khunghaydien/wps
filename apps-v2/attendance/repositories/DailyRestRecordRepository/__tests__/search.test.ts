import Api from '../../../../../__tests__/mocks/ApiMock';
import search from '../search';
import { defaultValue } from './mocks/search.mock';

it('should return dailyRestList', async () => {
  // Arrange
  const MockApi = Api.invoke as jest.Mock;
  MockApi.mockResolvedValueOnce(defaultValue);

  // Act
  const dailyRestList = await search({
    employeeId: 'employeeId',
    startDate: '2022-07-01',
    endDate: '2022-07-31',
  });

  // Assert
  expect(dailyRestList).toMatchSnapshot();
  expect(MockApi).toBeCalledTimes(1);
  expect(MockApi).toBeCalledWith({
    path: '/att/daily-rest/get',
    param: {
      empId: 'employeeId',
      startDate: '2022-07-01',
      endDate: '2022-07-31',
    },
  });
});
