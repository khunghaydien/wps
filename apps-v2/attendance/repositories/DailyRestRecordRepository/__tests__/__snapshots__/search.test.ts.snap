// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return dailyRestList 1`] = `
Array [
  Object {
    "recordDate": "2022-07-01",
    "restRecords": Array [
      Object {
        "id": "XXXXXXXXXXXXXXX",
        "outEndTime": 1200,
        "outRestTime": 120,
        "outStartTime": 1080,
        "restReasonCode": "TestCode",
        "restReasonId": "XXXXXXXXXXXXXXX",
        "restReasonName": "TestName",
      },
    ],
  },
]
`;
